controllers/controllers/resource/fetcher.go

```diff
@@ -430,6 +487,38 @@ func (r *CapiResourceFetcher) ExistingVSphereWorkerMachineConfig(ctx context.Con
 	return MapMachineTemplateToVSphereMachineConfigSpec(vsMachineTemplate, users)
 }
 
+func (r *CapiResourceFetcher) ExistingCloudStackDatacenterConfig(ctx context.Context, cs *anywherev1.Cluster, wnc anywherev1.WorkerNodeGroupConfiguration) (*anywherev1.CloudStackDatacenterConfig, error) {
```

Comment:
Ditto re unit tests?

Reply:
Looks like they're missing for this method, good catch. However, they seem to be missing across the board for all providers, so I'm inclined to create an issue about it for follow-up.

Reply:
I don't think existing code not having unit tests is justification for not writing unit tests. We'd end up merging a lot of untested code if we took that as precedent. 

An issue is required either way, no need to write missing unit tests.

Reply:
I agree. Created https://github.com/aws/eks-anywhere/issues/1644

Reply:
One contributing factor to that is it's difficult to mock the client.Reader dependency. It populates values by reference and doesn't actually return any values. This might require more effort

Reply:
Added unit tests here